course/src/isometry.cc

```diff
@@ -160,23 +153,9 @@ namespace ekumen
 
     Isometry operator*(const Isometry &i, const Isometry &i2)
     {
-        Isometry m;
-        Matrix3 p = i.matrix3_isometry;
-        Matrix3 q = i2.matrix3_isometry;
-
-        Vector3 a, b;
-
-        for (size_t i = 0; i < 3; ++i)
-        {
-            a = p.row(i);
-            for (size_t j = 0; j < 3; ++j)
-            {
-                b = q.col(j);
-                m.matrix3_isometry[i][j] = a.dot(b);
-            }
-        }
-        m.vector3_isometry = i.vector3_isometry + i2.vector3_isometry;
-        return m;
+        Matrix3 m1 = m1.mul(i.matrix3_isometry, i2.matrix3_isometry);
```

Comment:
As a side note, it's usually bad practice to use short variable names such as `i` and `i2`. Also, it's usually frown upon to use `i` for anything other than loop indexes, since that's the universally accepted use of `i`, `j` and `k`. Notice that there's people who would object even their use within this narrow context.

[Please read this](https://google.github.io/styleguide/cppguide.html#General_Naming_Rules)